### ISSUE TITLE
Add documentation on interacting with packages from Bioconductor

### ISSUE THREAD

**Issue by khughitt** [rstudio/renv#81]
From a quick look at the good, it appears that renv has some support for working with packages hosted on Bioconductor.

Attempting to initialize an environment containing such packages fails, however:

```
* Discovering package dependencies ... Done!
* Copying packages into the cache ... Done!
* Resolving missing dependencies  ... 
* Querying repositories for available source packages ... Done!
Retrieving 'http://lib.stat.cmu.edu/R/CRAN/src/contrib/foreach_1.4.4.tar.gz' ...
        OK [file is up to date]
Retrieving 'http://lib.stat.cmu.edu/R/CRAN/src/contrib/codetools_0.2-16.tar.gz' ...
        OK [file is up to date]
Retrieving 'http://lib.stat.cmu.edu/R/CRAN/src/contrib/iterators_1.0.10.tar.gz' ...
        OK [file is up to date]
Error: failed to retrieve package 'limma' (missing record)
Traceback (most recent calls first):
  9: stop(sprintf(fmt, ...), call. = call.)
  8: stopf(fmt, package)
  7: renv_retrieve_missing_record(package)
  6: records[[package]] %||% renv_retrieve_missing_record(package)
  5: renv_retrieve_impl(package)
  4: renv_retrieve(packages)
  3: renv_hydrate_resolve_missing(project, na)
  2: hydrate(project = project, library = library)
  1: renv::init()
```

Attempting to install the package via `renv::install()` results in a similar failure.

A work-around is to first install any needed packages manually using `BiocManager::install()`, and _then_ call `renv::init()`, at which point renv is able to detect the locally-installed packages and use them.

Is this something that should be supported? And if so, would it be possible to update the docs to indicate how? Otherwise, a note in the README.md on expected behavior / advice for working with Bioconductor, etc. packages might be helpful to include.

**Comment by kevinushey** [comment]
As a side note, this works for me if `limma` is installed in the user library. The problem here is that if `limma` is _not_ in the user library, then `renv` will assume the package is available on CRAN and try to retrieve it from there. Of course, it isn't, so the attempt to retrieve fails.

One workaround would be to explicitly load and assign the Bioconductor repositories before calling `renv::init()`, e.g.

```
> options(repos = BiocManager::repositories())
> renv::install("limma")
* Querying repositories for available binary packages ... Done!
* Querying repositories for available source packages ... Done!
Retrieving 'https://bioconductor.org/packages/3.8/bioc/bin/macosx/el-capitan/contrib/3.5/limma_3.38.3.tgz' ...
        OK [downloaded 2.6 Mb in 0.5 secs]
Installing limma [3.38.3] from CRAN ...
        OK (installed binary)
```

I'm not sure if this is the right solution in general, though.

**Comment by khughitt** [comment]
Thanks for the workarounds!

So it seems like a lot of these issues relate to me combining conda + renv, and as a result, going against some of the expectations of renv (e.g. that packages that are used will be available in the user library).

Since this seems like a pretty useful thing to do, however, in order to control the specific R version used on a system, I suspect others will end up running into similar issues down the road.

Initially, my thought was to do something like fallback on checking for packages in Bioconductor upon encountering the first missing package, or perhaps prompt the user to see if they want to enable bioconductor, or point renv to some additional library not currently included in `.libPaths()`.

A better approach, however, might be to explicitly check to see if renv is being called in a conda environment, and then adjust behavior accordingly.. e.g.

```r
conda_enabled <- Sys.getenv('CONDA_PREFIX') != ''
```

At this point, one could then check (for example) to see if `Sys.getenv("R_LIBS")`, etc. are in `.libPaths()`, and if not, prompt to user to see if they wish to include those additional paths to be searched.

**Comment by kevinushey** [comment]
How exactly are you initializing your Conda environments and launching R? (Just so I can try to test and replicate locally and better understand how these things differ in a Conda env)

**Comment by khughitt** [comment]
Sure thing. The steps I posted on #82 should walk you through the major steps. The only step I left out there is downloading and installing the latest Miniconda3:

```sh
wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh
sh Miniconda3-latest-Linux-x86_64.sh
```

Afterwards, activate conda for the shell as instructed and then you should be all set to create an new environment (and then activate it.)

At that point you can just use `R` to launch R; `$PATH` is updated by conda so that the environment-specific version is used.

From there you can try and restore the renv environment posted to the same issue, which includes a mix of CRAN / Bioconductor / Github packages.

**Comment by kevinushey** [comment]
I've decided to side-step the issue here a bit -- now, failures to find and install packages during `renv::init()` are no longer fatal. Instead, we will log a warning for the user:

```
> renv::init()
* Discovering package dependencies ... Done!
* Copying packages into the cache ... Done!
* Resolving missing dependencies  ...
* Querying repositories for available binary packages ... Done!
* Querying repositories for available source packages ... Done!
The following package(s) were not installed successfully:

        [missing]: failed to retrieve package 'missing' (missing record)

You may need to manually download and install these packages.

* Lockfile written to '~/scratch/bioc/renv.lock'.
* renv activated -- please restart the R session.
```

This way, one still gets a new `renv` project initialized, but gets informed that they're using a package that `renv` wasn't able to automatically find and install.

Let me know if this seems sensible for your use case.

**Comment by khughitt** [comment]
This seems like a reasonable approach. As long as _most_ of the environment initializes properly, it is not hard to follow that with a call `options(repos = BiocManager::repositories())` + manual installation of Bioconductor packages.
